vm-deployment/vm.bicep

```
@@ -0,0 +1,270 @@
+    name: 'HTTP'
+    protocol: 'TCP'
+    sourcePortRange: '*'
+    destinationPortRange: '80'
+    sourceAddressPrefix: '*'
+    destinationAddressPrefix: '*'
+    access: 'Allow'
+    priority: 320
+    direction: 'Inbound'
+  }
+  {
+    name: 'HTTPS'
+    protocol: 'TCP'
+    sourcePortRange: '*'
+    destinationPortRange: '443'
+    sourceAddressPrefix: '*'
+    destinationAddressPrefix: '*'
+    access: 'Allow'
+    priority: 340
+    direction: 'Inbound'
+  }
+]
+
+// Networking security group.
+resource networkSecurityGroup 'Microsoft.Network/networkSecurityGroups@2023-11-01' = {
+  name: networkSecurityGroups
+  location: location
+  properties: {
+    securityRules: [for rule in securityRules: {
+      name: rule.name
+      properties: {
+        protocol: rule.protocol
+        sourcePortRange: rule.sourcePortRange
+        destinationPortRange: rule.destinationPortRange
+        sourceAddressPrefix: rule.sourceAddressPrefix
+        destinationAddressPrefix: rule.destinationAddressPrefix
+        access: rule.access
+        priority: rule.priority
+        direction: rule.direction
+      }
+    }]
+  }
+}
+
+// Public IP address.
+resource publicIPAddresses_resource 'Microsoft.Network/publicIPAddresses@2023-11-01' = {
+  name: publicIPAddresses
+  location: location
+  sku: {
+    name: 'Standard'
+    tier: 'Regional'
+  }
+  zones: [
+    '1'
+  ]
+  properties: {
+    ipAddress: publicIPAddress
+    publicIPAddressVersion: 'IPv4'
+    publicIPAllocationMethod: 'Static'
+    idleTimeoutInMinutes: 4
+    ipTags: []
+  }
+}
+
+// Virtual network.
+resource virtualNetworks_resource 'Microsoft.Network/virtualNetworks@2023-11-01' = {
+  name: virtualNetworks
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        vnetAddressPrefix
+      ]
+    }
+    subnets: [
+      {
+        name: 'default'
+        properties: {
+          addressPrefix: subnetAddressPrefix
+          delegations: []
+          privateEndpointNetworkPolicies: 'Disabled'
+          privateLinkServiceNetworkPolicies: 'Enabled'
+        }
+      }
+    ]
+    virtualNetworkPeerings: []
+    enableDdosProtection: false
+  }
+}
+
+// Virtual machine.
+resource virtualMachines_resource 'Microsoft.Compute/virtualMachines@2024-03-01' = {
+  name: virtualMachines
+  location: location
+  properties: {
+    hardwareProfile: {
+      vmSize: vmSize
+    }
+    additionalCapabilities: {
+      hibernationEnabled: false
+    }
+    storageProfile: {
+      imageReference: {
+        publisher: 'canonical'
+        offer: '0001-com-ubuntu-server-focal'
+        sku: '20_04-lts-gen2'
+        version: 'latest'
+      }
+      osDisk: {
+        osType: 'Linux'
+        name: '${virtualMachines}_OsDisk_1_bcb7e14b83d24fc7a077197bdf465401'
+        createOption: 'FromImage'
+        caching: 'ReadWrite'
+        managedDisk: {
+          storageAccountType: 'Premium_LRS'
+        }
+        deleteOption: 'Delete'
+        diskSizeGB: 30
```

```suggestion
        diskSizeGB: diskSizeGB
```
Could we parametrize the disk size? For example:
```bicep
@description('Operating system disk size in Gigabytes.')
param diskSizeGB int = 30
```